scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3114,35 +3118,56 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
         name match {
           case q: Quasi if endTparamQuasi => q.become[Type.Param]
           case _ =>
-            val tparams = typeParamClauseOpt(ownerIsType = true, ctxBoundsAllowed = false)
+            val tparams = typeParamClauseOpt(
+              ownerIsType = true,
+              ctxBoundsAllowed = dialect.allowImprovedBoundsAndGivens
+            )
             val tbounds = typeBounds()
             val vbounds = new ListBuffer[Type]
             val cbounds = new ListBuffer[Type]
             if (ctxBoundsAllowed) {
               @inline
-              def getBound(): Type = {
-                next()
-                currToken match {
-                  case t: Ellipsis => ellipsis[Type](t, 1)
-                  case _ => typ()
-                }
+              def getBound(allowBoundsAlias: Boolean = false): Type = currToken match {
+                case t: Ellipsis => ellipsis[Type](t, 1)
+                case _ => contextBoundOrAlias(allowBoundsAlias)
               }
-              if (at[Viewbound]) {
+              if (acceptOpt[Viewbound]) {
                 if (!dialect.allowViewBounds) {
                   val msg = "Use an implicit parameter instead.\n" +
                     "Example: Instead of `def f[A <% Int](a: A)` " +
                     "use `def f[A](a: A)(implicit ev: A => Int)`."
                   syntaxError(s"View bounds are not supported. $msg", at = currToken)
                 }
-                doWhile(vbounds += getBound())(at[Viewbound])
+                doWhile(vbounds += getBound())(acceptOpt[Viewbound])
+              }
+              def addContextBound() = cbounds += getBound(allowBoundsAlias = true)
+              currToken match {
```

Comment:
if there are no other match cases, here's an alternative:
```
if (acceptOpt[Colon]) {
  def addContextBounds[T : ClassTag]: Unit =
    doWhile(addContextBound(allowBoundsAlias = false))(acceptOpt[T])
  if (acceptOpt[LeftBrace]) inBracesAfterOpen(addContextBounds[Comma])
  else addContextBounds[Colon]
}
```

Reply:
actually, could even inline the `addContextBound` method here as well, it's not used anywhere else. so
```
if (acceptOpt[Colon]) {
  def addContextBounds[T : ClassTag]: Unit =
    doWhile(cbounds += getBound(allowBoundsAlias = true))(acceptOpt[T])
  if (acceptOpt[LeftBrace]) inBracesAfterOpen(addContextBounds[Comma])
  else addContextBounds[Colon]
}
```
and `addContextBound` didn't take a parameter :)

Reply:
Thanks! Good idea, changed